File: momentra/momentra/Shell/Setup/SetupPreviewCard.swift
import SwiftUI

struct SetupPreviewBar: Identifiable {
    let id = UUID()
    let label: String
    let progress: Double
    let color: Color
}

struct SetupPreviewCard: View {
    let title: String
    var subtitle: String? = nil
    var bullets: [String] = []
    var bars: [SetupPreviewBar] = []
    var badge: String? = nil

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            HStack {
                Text(title)
                    .font(.system(size: 14, weight: .semibold))
                    .foregroundStyle(SetupTokens.textPrimary)
                Spacer()
                if let badge {
                    Text(badge)
                        .font(.system(size: 10, weight: .bold))
                        .foregroundStyle(SetupTokens.brandPrimary)
                        .padding(.horizontal, 8)
                        .padding(.vertical, 4)
                        .background(SetupTokens.accentPurple.opacity(0.2), in: RoundedRectangle(cornerRadius: 8))
                }
            }

            if let subtitle {
                Text(subtitle)
                    .font(.system(size: 12))
                    .foregroundStyle(SetupTokens.textSecondary)
            }

            ForEach(bars) { bar in
                VStack(alignment: .leading, spacing: 4) {
                    Text(bar.label.uppercased())
                        .font(.system(size: 10, weight: .bold))
                        .foregroundStyle(SetupTokens.textSecondary)
                    GeometryReader { geo in
                        ZStack(alignment: .leading) {
                            RoundedRectangle(cornerRadius: 3)
                                .fill(Color.white.opacity(0.08))
                            RoundedRectangle(cornerRadius: 3)
                                .fill(bar.color)
                                .frame(width: geo.size.width * min(max(bar.progress, 0), 1))
                        }
                    }
                    .frame(height: 6)
                }
            }

            ForEach(bullets, id: \.self) { bullet in
                Text("• \(bullet)")
                    .font(.system(size: 12))
                    .foregroundStyle(SetupTokens.textSecondary)
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(16)
        .background(SetupTokens.surfaceCard, in: RoundedRectangle(cornerRadius: 16))
        .overlay(RoundedRectangle(cornerRadius: 16).stroke(SetupTokens.borderSubtle, lineWidth: 1))
    }
}

struct SetupMissionCard: View {
    let title: String
    let bodyText: String

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            Text(title.uppercased())
                .font(.system(size: 11, weight: .bold))
                .kerning(1)
                .foregroundStyle(SetupTokens.brandPrimary)
            Text(bodyText)
                .font(.system(size: 14))
                .foregroundStyle(SetupTokens.textPrimary)
                .fixedSize(horizontal: false, vertical: true)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(16)
        .background(SetupTokens.surfaceCard, in: RoundedRectangle(cornerRadius: 16))
        .overlay(RoundedRectangle(cornerRadius: 16).stroke(SetupTokens.borderSubtle, lineWidth: 1))
    }
}

File: apk/app/src/main/java/com/example/momentra/ui/setup/SetupPreviewCard.kt
package com.example.momentra.ui.setup

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

data class SetupPreviewBar(
    val label: String,
    val progress: Float,
    val color: Color,
)

@Composable
fun SetupPreviewCard(
    title: String,
    subtitle: String? = null,
    bullets: List<String> = emptyList(),
    bars: List<SetupPreviewBar> = emptyList(),
    badge: String? = null,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(16.dp))
            .background(SetupTokens.SurfaceCard)
            .border(1.dp, SetupTokens.BorderSubtle, RoundedCornerShape(16.dp))
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(title, color = SetupTokens.TextPrimary, fontWeight = FontWeight.SemiBold, fontSize = 14.sp)
            if (badge != null) {
                Text(
                    badge,
                    color = SetupTokens.BrandPrimary,
                    fontSize = 10.sp,
                    fontWeight = FontWeight.Bold,
                    modifier = Modifier
                        .clip(RoundedCornerShape(8.dp))
                        .background(SetupTokens.AccentPurple.copy(alpha = 0.2f))
                        .padding(horizontal = 8.dp, vertical = 4.dp),
                )
            }
        }
        if (subtitle != null) {
            Text(subtitle, color = SetupTokens.TextSecondary, fontSize = 12.sp)
        }
        bars.forEach { bar ->
            Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
                Text(bar.label, color = SetupTokens.TextSecondary, fontSize = 10.sp, fontWeight = FontWeight.Bold)
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(6.dp)
                        .clip(RoundedCornerShape(3.dp))
                        .background(Color.White.copy(alpha = 0.08f)),
                ) {
                    Box(
                        modifier = Modifier
                            .fillMaxWidth(bar.progress.coerceIn(0f, 1f))
                            .height(6.dp)
                            .clip(RoundedCornerShape(3.dp))
                            .background(bar.color),
                    )
                }
            }
        }
        bullets.forEach { bullet ->
            Text("• $bullet", color = SetupTokens.TextSecondary, fontSize = 12.sp)
        }
    }
}

@Composable
fun SetupMissionCard(
    title: String,
    body: String,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(16.dp))
            .background(SetupTokens.SurfaceCard)
            .border(1.dp, SetupTokens.BorderSubtle, RoundedCornerShape(16.dp))
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Text(title, color = SetupTokens.BrandPrimary, fontSize = 11.sp, fontWeight = FontWeight.Bold, letterSpacing = 1.sp)
        Text(body, color = SetupTokens.TextPrimary, fontSize = 14.sp, lineHeight = 20.sp)
    }
}

@Composable
fun SetupFieldCard(
    label: String,
    value: String,
    icon: String? = null,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(SetupTokens.SurfaceCard)
            .border(1.dp, SetupTokens.BorderSubtle, RoundedCornerShape(12.dp))
            .padding(horizontal = 14.dp, vertical = 14.dp),
        horizontalArrangement = Arrangement.spacedBy(10.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        if (icon != null) {
            Text(icon, fontSize = 16.sp)
        }
        Column(modifier = Modifier.weight(1f)) {
            Text(label, color = SetupTokens.TextSecondary, fontSize = 11.sp)
            Text(value.ifBlank { "—" }, color = SetupTokens.TextPrimary, fontSize = 14.sp)
        }
    }
}
